deeplay/components/rnn.py

```diff
@@ -1,147 +1,279 @@
+    - dropout (float): If non-zero, introduces a dropout layer on the outputs of each RNN layer except the last layer. (Default: 0.0)
+    """
+    
+    in_features: Optional[int] 
+    hidden_features: Sequence[Optional[int]] 
+    rnn_type: Optional[str]
+    bidirectional: bool
+    batch_first: bool
+    dropout: float
+    blocks: LayerList[Layer]
+    
+    @property
+    def input(self):
+        """Return the input layer of the network. Equivalent to `.blocks[0]`."""
+        return self.blocks[0]
 
-    Constraints
-    -----------
-    - input shape: (batch_size, seq_len, input_size)
-    - output shape: (batch_size, seq_len, hidden_size * num_directions)
+    @property
+    def hidden(self):
+        """Return the hidden layers of the network. Equivalent to `.blocks[:-1]`"""
+        return self.blocks[:-1]
 
-    Evaluation
-    ----------
-    >>> output, hidden = self.rnn(x)
-    >>> return output, hidden
+    @property
+    def output(self):
+        """Return the last layer of the network. Equivalent to `.blocks[-1]`."""
+        return self.blocks[-1]
 
-    Examples
-    --------
-    >>> # With default configuration
-    >>> rnn = BaseRNN(Config().update(input_size=my_input_size, hidden_size=my_hidden_size))
-    >>> # With custom number of layers and dropout
-    >>> rnn = BaseRNN(Config().update(input_size=my_input_size, hidden_size=my_hidden_size, num_layers=2, dropout=0.1, rnn_type='GRU'))
+    @property
+    def layer(self) -> LayerList[Layer]:
+        """Return the layers of the network. Equivalent to `.blocks.layer`."""
+        return self.blocks.layer
 
-    Return Values
-    -------------
-    The forward method returns a tuple of two elements:
-    - outputs: the output features `h` from the last layer of the RNN, for each timestep
-    - hidden: the hidden state for the last timestep for each layer, `h_n`
+    @property
+    def activation(self) -> LayerList[Layer]:
+        """Return the activations of the network. Equivalent to `.blocks.activation`."""
+        return self.blocks.activation
 
-    Additional Notes
-    ----------------
-    The `Config` class is used for configuring the RNN. For more details refer to [Config Documentation](#).
-    """
+    @property
+    def normalization(self) -> LayerList[Layer]:
+        """Return the normalizations of the network. Equivalent to `.blocks.normalization`."""
+        return self.blocks.normalization
 
-    @staticmethod
-    def defaults():
-        return (
-            Config()
-            .input_size(None)  # Expected to be set by user
-            .hidden_size(256)
-            .num_layers(1)
-            .dropout(0.1)
-            .rnn_type('LSTM')  # Can be 'LSTM' or 'GRU'
-        )
+    def __init__(
+        self, 
+        in_features: int, 
+        hidden_features: Optional[int] = 256, 
+        num_layers: Optional[int] = 1,
```

Comment:
Can we do the same thing as in cnn and mlp? as in, instead of specifying num_layers, we give `hidden_features` as a list of the number of features per layer?